pkg/ctrlz/ctrlz.go

```diff
@@ -121,20 +121,20 @@ func RegisterTopic(t fw.Topic) {
 	allTopics = append(allTopics, t)
 }
 
-var listeningTestProbe func()
-
 // Run starts up the ControlZ listeners.
 //
 // ControlZ uses the set of standard core topics, the
 // supplied custom topics, as well as any topics registered
 // via the RegisterTopic function.
-func Run(o *Options, customTopics []fw.Topic) {
-	shutdown.Add(1)
-	defer shutdown.Done()
-
+func Run(o *Options, customTopics []fw.Topic) (*Server, error) {
 	if o.Port == 0 {
 		// disabled
-		return
+		s := &Server{}
```

Comment:
There is a re-entrancy race here I think. (Not sure how important that is).

However, given that Run does not block caller, it looks like you can handle state-transition protection with a basic lock, and keep using the same wg approach to ensure that the background go routine exits by the time Close returns control to the caller?

Reply:
I don't see the reentrancy issue, can you clarify?